detox/e2e/test/messaging/permalink.e2e.js

```diff
@@ -0,0 +1,150 @@
+        await postMessage(message1);
+
+        // * Check that message is successfully posted
+        await expect(element(by.text(message1))).toExist();
+
+        // # Post Private Channel 2's POST Permalink
+        const message2 = `${serverUrl}/${testTeam.name}/pl/${post.id}`;
+        await postMessage(message2);
+
+        // * Check that message is successfully posted
+        await expect(element(by.text(message2))).toExist();
+
+        // # Logout and login as sysadmin
+        await logout();
+        await ChannelScreen.open({
+            username: adminUsername,
+            password: adminPassword,
+        });
+
+        // * Verify channel screen is visible
+        await ChannelScreen.toBeVisible();
+
+        // # Go to the team
+        await openTeamSidebar();
+        await getTeamByDisplayName(testTeam.display_name).tap();
+
+        // # Press on message 1
+        await tapLink(message1);
+
+        // # Press on Join button
+        await joinChannel();
+
+        // * Confirm joining the "private channel 1"
+        await expect(ChannelScreen.channelIntro).toHaveText('Beginning of ' + privateChannel1Name);
+
+        // # Go to Townsquare
+        await gotoChannel('Town Square');
+
+        // # Press on message 2
+        await tapLink(message2);
+
+        // # Press on Join button
+        await joinChannel();
+
+        // * Verify permalink post list has the message
+        await PermalinkScreen.toBeVisible();
+        await expect(element(by.text(post.message))).toBeVisible();
+
+        // # Jump to recent messages
+        await PermalinkScreen.jumpToRecentMessages();
+
+        // * Verify user is on channel where message is posted
+        await expect(ChannelScreen.channelIntro).toHaveText('Beginning of ' + privateChannel2Name);
+    });
+});
+
+async function createPrivateChannel(channelName) {
+    // # Open Mainside bar and press on private channels more button
+    await ChannelScreen.openMainSidebar();
+    await MainSidebar.openCreatePrivateChannelButton.tap();
+
+    // * Verify create channel screen is visible
+    await CreateChannelScreen.toBeVisible();
+    await expect(element(by.text('New Private Channel'))).toBeVisible();
+
+    // # Fill the data and create a private channel
+    await CreateChannelScreen.nameInput.typeText(channelName);
+    await CreateChannelScreen.createButton.tap();
+
+    // * Expect a redirection to the created channel
+    await expect(ChannelScreen.channelIntro).toHaveText('Beginning of ' + channelName);
+}
+
+async function gotoChannel(name) {
+    await ChannelScreen.openMainSidebar();
+    const channelItem = MainSidebar.getChannelByDisplayName(name);
+    await channelItem.tap();
+    await expect(ChannelScreen.channelNavBarTitle).toHaveText(name);
+}
+
+async function joinChannel() {
+    const button = isAndroid() ? element(by.text('JOIN')) : element(by.label('Join')).atIndex(0);
+    await button.tap();
+}
```

Comment:
Please add entries in `support/ui/component/alert.js`,
```
// alert titles
...
joinPrivateChannelTitle = isAndroid() ? element(by.text('Join private channel')) : element(by.label('Join private channel')).atIndex(0);

// alert buttons
...
joinButton = isAndroid() ? element(by.text('JOIN')) : element(by.label('Join')).atIndex(0);
```

```suggestion
async function joinPrivateChannel() {
    await expect(Alert.joinPrivateChannelTitle).toBeVisible();
    await Alert.joinButton.tap();
}
```